### PR TITLE
Add GeotrellisRasterSource

### PR DESCRIPTION
**Pull request by pcaisse**
## Overview

Read in a GT catalog by layer ID.

## Notes

Some tests borrowed from `GeoTiffRasterSource`. Resampling and reprojection to come in subsequent PRs.

### CODE REVIEW

File: vlm/src/main/scala/geotrellis/contrib/vlm/GeotrellisRasterSource.scala [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+/**
+  * Note: GeoTrellis AttributeStore does not store the band count for the layers by default,
+  *       thus they need to be provided from application configuration.
+  *
+  * @param uri geotrellis catalog uri
+  * @param layerId source layer from above catalog
+  * @param bandCount number of bands for each tile in above layer
+  */
+case class GeotrellisRasterSource(uri: String, layerId: LayerId, bandCount: Int = 1) extends RasterSource {
```
Comment (echeipesh) [inline vlm/src/main/scala/geotrellis/contrib/vlm/GeotrellisRasterSource.scala]: For consistency this should move into its own package.
I propose `goetrellis.contrib.vlm.avro`

Comment (pcaisse) [reply vlm/src/main/scala/geotrellis/contrib/vlm/GeotrellisRasterSource.scala]: Sure. Should I also move `TestCatalog.scala` into `goetrellis.contrib.vlm.avro` then?

Comment (echeipesh) [reply vlm/src/main/scala/geotrellis/contrib/vlm/GeotrellisRasterSource.scala]: Thats a good point, it should go there.

**Comment by echeipesh** [comment]
Looks like the test is failing with `pure virtual method called` error at cleanup time. This looks like its related to GDAL cleanup and and not anything going on in this PR.

**Pull request merged**